flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/RowDataFileScanTaskReader.java

```diff
@@ -83,6 +95,12 @@ public RowDataFileScanTaskReader(Schema tableSchema, Schema projectedSchema,
       iterable = CloseableIterable.transform(iterable, rowDataProjection::wrap);
     }
 
+    if (projectedFields != null) {
+      RowDataNestProjection rowDataProjection = RowDataNestProjection.create(tableSchema, projectedSchema,
+          projectedFields);
+      iterable = CloseableIterable.transform(iterable, rowDataProjection::wrap);
+    }
```

Comment:
Why do we need to project the iterable<RowData> twice ( we have already projected the fields in line95, so why need this extra projection...)

Reply:
Line95 is used to remove the extra column fields from the deleted file, but does not extract the nested fields.


![image](https://user-images.githubusercontent.com/59213263/157242838-69ff434c-b6ea-4b6c-b982-658bb98ed514.png)

Reply:
I optimized this part and unified it. Please take a look. Thank you.